packages/infra/infra-core/src/lib/patterns/ecr-sync/lambda/get-image-tags-handler.ts

```diff
@@ -0,0 +1,134 @@
+    let ecrImageName = image.imageName;
+
+    if (repoPrefix.length > 0) {
+      ecrImageName = `${repoPrefix}/${ecrImageName}`;
+    }
+
+    const ecrImageTags = await getEcrImageTags(ecrImageName);
+    console.debug(`ECR images for ${image.imageName}: \n${ecrImageTags.map(t => `${t.tag} (${t.digest})`).join('\n')}`);
+
+    // List all image tags in Docker
+    const dockerImageTags = await getDockerImageTags(image.imageName);
+    console.debug(`Docker images for ${image.imageName}: ${dockerImageTags.map(t => `${t.tag} (${t.digest})`).join('\n')}`);
+
+    let missingImageTags = await filterTags(dockerImageTags, ecrImageTags, image);
+    let orderedImageTags = await orderTags(missingImageTags);
+    buildTriggerFile += await formatTriggerLines(image, orderedImageTags, repoPrefix, accountId, region);
+  }));
+
+  console.info(`Images to sync:\n${buildTriggerFile}`);
+
+  if (buildTriggerFile === '') return;
+
+  const stream = await zipToFileStream(buildTriggerFile);
+  await uploadToS3(process.env['BUCKET_NAME']!, 'images.zip', stream);
+}
+
+export async function orderTags(imageTags: ContainerImage[]) {
+  return imageTags.sort((a, b) => a.lastUpdated.localeCompare(b.lastUpdated));
+}
+
+export async function formatTriggerLines(image: Image, missingImageTags: ContainerImage[], repoPrefix: string, accountId: string, region: string) {
+
+  let buildTriggerFile: string = '';
+
+  missingImageTags.forEach(t => {
+    const ecrImageName = (repoPrefix) ? `${repoPrefix}/${image.imageName}` : image.imageName;
+    buildTriggerFile += `${image.imageName},${accountId}.dkr.ecr.${region}.amazonaws.com/${ecrImageName},${t.tag}\n`;
+  });
+
+  return buildTriggerFile;
+}
+
+export async function filterTags(dockerImageTags: ContainerImage[], ecrImageTags: ContainerImage[], image: Image) {
+
+  let missingImageTags: ContainerImage[] = dockerImageTags.filter(dockerImage => {
+    return ecrImageTags.filter(ecrImage => ecrImage.tag === dockerImage.tag && ecrImage.digest === dockerImage.digest).length === 0;
+  });
+
+  return missingImageTags.filter(t => {
+
+    // If nothing is defined allow all
+    if (image.includeTags === undefined && image.excludeTags === undefined) {
+      return true;
+    }
+
+    // Deny if tag matches `excludeTags`
+    if (image.excludeTags !== undefined) {
+      if (image.excludeTags.some(excludeTag => t.tag.match(excludeTag))) {
+        return false;
+      };
+    }
+
+    // Allow if tag matches `includeTags`
+    if (image.includeTags !== undefined) {
+      if (image.includeTags.some(includeTag => t.tag.match(includeTag))) {
+        return true;
+      }
+    }
+
+    // If only excludeTags is defined all others should be included
+    if (image.includeTags === undefined && image.excludeTags !== undefined) {
+      return true;
+    }
+
+    return false;
+  });
+}
+
+async function uploadToS3(bucket: string, key: string, stream: PassThrough) {
+
+  const params: PutObjectRequest = {
+    ACL: 'private',
```

Comment:
ACL parameter is probably redundant in our case bc ACLs are disabled by default for new buckets since April 2023

> ACLs Disabled – The Bucket owner enforced setting will be enabled for newly created buckets, making bucket ACLs and object ACLs ineffective, and ensuring that the bucket owner is the object owner no matter who uploads the object.

https://aws.amazon.com/blogs/aws/heads-up-amazon-s3-security-changes-are-coming-in-april-of-2023/

And setting ACL for objects inside bucket with owner enforced settings might cause errors

> If your bucket uses the bucket owner enforced setting for S3 Object Ownership, you must use policies to grant access to your bucket and the objects in it. With the bucket owner enforced setting enabled, requests to set access control lists (ACLs) or update ACLs fail and return the AccessControlListNotSupported error code.

https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html#CannedACL